### ISSUE TITLE
MultiSim shrink() fails with uninitialized sim

### ISSUE THREAD

**Issue by cliffckerr** [starsimhub/starsim#825]
```py
import sciris as sc
import starsim as ss

sim = ss.Sim(diseases='sis', networks='random')
# sim.init() # required to run

msim = ss.MultiSim(base_sim=sim)
msim.run()
msim.shrink() # error
```

Also: add a `keep` kwarg to avoid shrinking/checking user-specified modules.

**Comment by cliffckerr** [comment]
Closed by #831

**Comment by cliffckerr** [comment]
But see also #832
